src/lib/stitching/causality/stitching.ts

```diff
@@ -209,29 +212,94 @@ export const causalityStitchingEnvironment = ({
           },
         },
         watchedLotConnection: {
-          resolve: async (_parent, _args, { saleArtworksAllLoader }) => {
-            const watchedSaleArtworksReq = await saleArtworksAllLoader({
+          resolve: async (_parent, args, context, _info) => {
+            const { saleArtworksAllLoader, causalityLoader } = context
+            // fetch sale artworks from gravity
+            const { first = 10, ...rest } = args
```

Comment:
Fetching a default of 10 watched sale artworks from gravity. We define this in the stitching file rather than our local schema because the new `Lot` type relies on a causality type and we cannot reference it until the causality schema has been stitched in - for more context and approach see #2925.

How this resolver works:
Watched lots are determined by gravity, their states are fetched from causality, and they are combined, filtering out any sale artworks with missing lot states (this should never happen in production but is common in staging due to the presence of copied data that has not been synced to causality).

See all of this logic in the following lines of this resolver function.